sessions/backend-update-delete/demo-end/pages/api/jokes/[id].js

```diff
@@ -0,0 +1,29 @@
+import dbConnect from "../../../db/connect";
+import Joke from "../../../db/models/Joke";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  const { id } = request.query;
+
+  if (request.method === "GET") {
+    const joke = await Joke.findById(id);
+
+    if (!joke) {
+      return response.status(404).json({ status: "Not Found" });
+    }
+
+    response.status(200).json(joke);
+  }
+
+  if (request.method === "PUT") {
+    const jokeToUpdate = await Joke.findByIdAndUpdate(id, {
+      $set: request.body,
```

Comment:
I think you need to parse the body first: `JSON.parse(request.body)`

Reply:
This  made me curious, so I tested it.  🕵️‍♀️  

Because we use the `$set` expression we don't have to parse the body. 
An alternative solution could be:

```javascript
  if (request.method === "PUT") {
    const data = JSON.parse(request.body);
    const jokeToUpdate = await Joke.findByIdAndUpdate(id, data);
    return response.status(200).json(jokeToUpdate);
  }
```
I can't judge nor decide which approach is the better 🫡 right 🤗 nicer 😎 cooler one. Do you?

Reply:
Interesting! thats actually quite useful.

Reply:
Hmm, I don't see much difference in complexity. cant decide

Reply:
I'll decide :D We'll keep it as is for the time being, as it was a recommendation by former employees.